### PR TITLE
Determinant Function By Xiaochen Xue and Yutong Su

### PR DESCRIPTION
**Pull request by LetUsWorship**

### CODE REVIEW

File: 02-library/cs506/determinant.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+def determinant(A):
+
+    n = len(A)
+    AM = copy_matrix(A)
```
Comment (gallettilance) [inline 02-library/cs506/determinant.py]: where is this function defined?

File: 02-library/cs506/determinant.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+    for fd in range(n):  
+        if AM[fd][fd] == 0:
```
Comment (gallettilance) [inline 02-library/cs506/determinant.py]: why?

Comment (LetUsWorship) [reply 02-library/cs506/determinant.py]: Hey Prof. Lance, I have found this implementation online, and found out this implementation does not make sense here. So I just wrote an easy to understand, recursive code and committed and updated on the same file. Please feel free to give me suggestions of the new one!

**Comment by LetUsWorship** [comment]
I think you have to merge my code first before I can submit a new request...

My bad on uploading lecture notes onto the same PR sorry!

**Pull request closed without merging**